File: model/especialista.py
from utils.db import db
from dataclasses import dataclass

@dataclass
class Especialista(db.Model):
    __tablename__ = 'especialista'
    id_especialista = db.Column(db.Integer, primary_key=True, autoincrement=True)
    id_persona = db.Column(db.Integer, db.ForeignKey('persona.id_persona'))
    codigo_especialista = db.Column(db.String(8))
    especialidad = db.Column(db.String(50))
    experiencia = db.Column(db.Integer)

    persona = db.relationship('Persona', backref='especialista')

    def __init__(self, id_persona, codigo_especialista, especialidad, experiencia):
        self.id_persona = id_persona
        self.codigo_especialista = codigo_especialista
        self.especialidad = especialidad
        self.experiencia = experiencia

File: v2-microservicios/sisvita-app/auth-server/src/main/java/sm/dsw/sisvita/auth/model/Especialista.java
package sm.dsw.sisvita.auth.model;

import jakarta.persistence.*;
import lombok.*;


@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "especialista")
public class Especialista {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_especialista;

    @OneToOne
    @JoinColumn(name = "id_persona", referencedColumnName = "id_persona")
    private Persona persona;

    @Column(name = "codigo_especialista", unique = true)
    private String codigo_especialista;

    @Column(name = "especialidad")
    private String especialidad;

    @Column(name = "experiencia")
    private String experiencia;

}

